src/serializer.rs

```diff
@@ -95,7 +95,7 @@ impl Header {
 impl SdPayload {
     /// Serializes the payload into a writer.
     pub fn to_vec(&self) -> Vec<u8> {
-        let mut buf = Vec::with_capacity(self.len());
+        let mut buf = Vec::with_capacity(self.length());
```

Comment:
Please consider https://doc.rust-lang.org/std/index.html?search=len vs https://doc.rust-lang.org/std/index.html?search=lenght

Reply:
imo the problem is that "len" implies "num of elements" (eg see: len_without_is_empty clippy lint), but actually we deal with the "size" of an item here.

maybe we should change:
```
impl SdPayload {
    pub fn length(&self) -> usize {
        let mut len: usize = SD_HEADER_LENGTH;
```

to something like:
```
impl SdPayload {
    pub fn size(&self) -> usize {
        let mut len: usize = SD_HEADER_SIZE;
```

btw: https://doc.rust-lang.org/std/index.html?search=size

Reply:
Think we have a "num of elements" situation? Bytes, entries etc...

Reply:
Hm. Unsure. I'm not so deep in the SomeIP spec. Think the `length` field includes the 8 bytes after this field as well. 

```rust
/// Raw length field
pub fn length(&self) -> u32 {
}

/// Payload length
pub fn payload_len(&self) -> usize {
   // something like self.length - 8 if applicable?
}
```

Reply:
done.